std/encoding/name_pattern.go

```diff
@@ -67,8 +67,8 @@ func (n Name) EncodingLength() int {
 // Clone returns a deep copy of a Name
 func (n Name) Clone() Name {
 	ret := make(Name, len(n))
-	for i, c := range n {
-		ret[i] = c.Clone()
+	for i := range n {
+		ret[i] = n[i].Clone()
```

Comment:
What's the difference here?

Reply:
Related to my comment on `PrefixHash`. The second element in range loop is byval, not ref